### ISSUE TITLE
False positive in Lint/SelfAssignment

### ISSUE THREAD

**Issue by ccutrer** [rubocop/rubocop#12435]
## Expected behavior

A self assignment where the "read" call has parameters should not trigger an offense.

## Actual behavior

An offense is triggered, when a parameter is passed. The parameter would imply that this is not a simple reader method, and may be doing some sort of calculation.

## Steps to reproduce the problem

```ruby
# frozen_string_literal: true

f.full_name = f.full_name(true)
```

`rubocop test.rb`

## RuboCop version

```
% rubocop -V
1.58.0 (using Parser 3.2.2.4, rubocop-ast 1.30.0, running on ruby 3.1.3) [arm64-darwin21]
```
